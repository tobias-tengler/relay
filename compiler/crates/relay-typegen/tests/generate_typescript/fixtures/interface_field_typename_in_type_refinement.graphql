fragment b on Query {
  node(id: "") {
    ... on User {
      name
    }
    ... on Page {
      __typename
      doesViewerLike
    }
  }
}