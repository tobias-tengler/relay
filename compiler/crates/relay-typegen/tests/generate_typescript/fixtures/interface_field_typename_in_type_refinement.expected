==================================== INPUT ====================================
fragment b on Query {
  node(id: "") {
    ... on User {
      name
    }
    ... on Page {
      __typename
      doesViewerLike
    }
  }
}
==================================== OUTPUT ===================================
import { FragmentRefs } from "relay-runtime";
export type b$data = {
  readonly node: {
    readonly __typename: "Page";
    readonly doesViewerLike: boolean | null | undefined;
  } | {
    // This will never be '%other', but we need some
    // value in case none of the concrete values match.
    readonly __typename: "%other";
  } | {
    readonly name: string | null | undefined;
  } | null | undefined;
  readonly " $fragmentType": "b";
};
export type b$key = {
  readonly " $data"?: b$data;
  readonly " $fragmentSpreads": FragmentRefs<"b">;
};
